src/test/java/com/vinylteam/vinyl/web/servlets/SignOutServletTest.java

```diff
@@ -12,25 +12,39 @@
 import static org.mockito.Mockito.*;
 
 class SignOutServletTest {
+    HttpServletRequest mockedRequest = mock(HttpServletRequest.class);
+    HttpServletResponse mockedResponse = mock(HttpServletResponse.class);
+    HttpSession mockedHttpSession = mock(HttpSession.class);
+    SignOutServlet signOutServlet = new SignOutServlet();
+    InOrder inOrderRequest = inOrder(mockedRequest);
+    InOrder inOrderResponse = inOrder(mockedResponse);
     @Test
-    @DisplayName("Checks if all right methods are called")
-    void doGetTest() throws IOException {
+    @DisplayName("Checks if all right methods are called & existing session was deleted")
+    void doGetWithExistingSessionTest() throws IOException {
         //prepare
-        HttpServletRequest mockedRequest = mock(HttpServletRequest.class);
-        HttpServletResponse mockedResponse = mock(HttpServletResponse.class);
-        HttpSession mockedHttpSession = mock(HttpSession.class);
-        SignOutServlet signOutServlet = new SignOutServlet();
-        InOrder inOrderRequest = inOrder(mockedRequest);
-        InOrder inOrderResponse = inOrder(mockedResponse);
-
-        when(mockedRequest.getSession()).thenReturn(mockedHttpSession);
+        when(mockedRequest.getSession(false)).thenReturn(mockedHttpSession);
         //when
         signOutServlet.doGet(mockedRequest, mockedResponse);
         //then
         inOrderResponse.verify(mockedResponse).setContentType("text/html;charset=utf-8");
-        inOrderRequest.verify(mockedRequest).getSession();
+        inOrderRequest.verify(mockedRequest).getSession(false);
         verify(mockedHttpSession).invalidate();
         inOrderResponse.verify(mockedResponse).setStatus(HttpServletResponse.SC_OK);
         inOrderResponse.verify(mockedResponse).sendRedirect("/");
     }
+
+    @Test
+    @DisplayName("Checks if all right methods are called & session doesn't exist")
+    void doGetWithNotExistingSessionTest() throws IOException {
+        //prepare
+        when(mockedRequest.getSession(false)).thenReturn(null);
+        //when
+        signOutServlet.doGet(mockedRequest, mockedResponse);
+        //then
+        inOrderResponse.verify(mockedResponse).setContentType("text/html;charset=utf-8");
+        inOrderRequest.verify(mockedRequest).getSession(false);
+        verify(mockedHttpSession, times(0)).invalidate();
+        inOrderResponse.verify(mockedResponse).setStatus(HttpServletResponse.SC_OK);
+        inOrderResponse.verify(mockedResponse).sendRedirect("/");
```

Comment:
Reusing same mock with inOrder is not secure.
(1)mockedResponse.setStatus(...);
(2)mockedResponse.sendRedirect(...);
(3)inOrderResponse.verify(...).setStatus(...);
(4)inOrderResponse.verify(...).sendRedirect(...);
(5)mockedResponse.setStatus(...);
(6)mockedResponse.sendRedirect(...);
(7)inOrderResponse.verify(...).setStatus(...);
(8)inOrderResponse.verify(...).sendRedirect(...);

The same as:

(1)mockedResponse.setStatus(...);
(2)mockedResponse.sendRedirect(...);
(5)mockedResponse.setStatus(...);
(6)mockedResponse.sendRedirect(...);
(3)inOrderResponse.verify(...).setStatus(...);
(4)inOrderResponse.verify(...).sendRedirect(...);
(7)inOrderResponse.verify(...).setStatus(...);
(8)inOrderResponse.verify(...).sendRedirect(...);

It has a potential to jumble tests up. Better to have @BeforeEach that nulls them and reassigns new mocks.